bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/ProgressCallback.java

```diff
@@ -74,4 +75,14 @@
      * Callback operation to indicate that the firmware update was canceled.
      */
     void canceled();
+    
+    /**
+     * Callback operation to indicate that the percentage progress of the firmware update has changed. 
+     * 
+     * @param progress the progress between 0 and 100
+     * @param silent if true EventSubscriber want get informed 
+     * 
+     * @throws IllegalArgumentException if given progress is < 0 or > 100
+     */
+    void setPercentageProgress(int progress, boolean silent); 
```

Comment:
I would prefer if the silent parameter is not part of the API... there might be several requirements how the percentage update is to be used by the solutions based on ESH... so if we require such a feature then we should introduce a more generic mechanism to prevent sending of progress update events